backend/src/v1/handlers/authHandler.ts

```diff
@@ -3,38 +3,55 @@ import User from "../classes/Auth";
 import ApiError from "../classes/ApiError";
 
 const authHandler = {
-  /* read: async (req: Request, res: Response) => {
-    const id = req.params.id;
-    // const userid = req.apiSession.userid;
-
-    if (id) {
-      if (id == id) {
-        const user = new User();
-        try {
-          const userData = await user.read(id);
+  read: async (req: Request, res: Response) => {
+    const address = req.params.publicAddress;
+    
+    if (address) {
+      const user = new User();
+      try {
+        const userData = await user.find(address);
+        if (userData) {
           res.status(200).send(userData);
-        } catch (reason) {
-          res.status(400).send(new ApiError(400, reason));
+        } else {
+          res
+            .status(403)
+            .send(new ApiError(403, "Access denied, userid does not match"));
         }
-      } else {
-        res.status(403).send(new ApiError(403, "Access denied, userid does not match"));
+      } catch (reason: any) {
+        res.status(400).send(new ApiError(400, reason));
       }
+    } else {
+      res.status(400).send(new ApiError(400, "Address is required"));
     }
-  }, */
+  },
 
   create: async (req: Request, res: Response) => {
-    /*   const user = new User();
-      user.set(req.body); // should be a user object
-  
-      try {
-        const result = await user.create();
-        res.status(200).send(result);
-      } catch (reject) {
-        res.status(400).send(new ApiError(400, reject));
-      } */
-  },
+    const {
+      email,
+      first_name,
+      last_name,
+      public_address,
+      service_provider_name,
+    } = req.body;
+    const user = new User();
 
+    try {
+      const result = await user.createUser({
+        email,
```

Comment:
You can also use the .set() function to set the user, to avoid passing in the whole object through params
user.set(req.body); // should be a user object

Reply:
oh got it, can be good, I m using the fields props directly because to not have the capability to send any weird or not needed data from the client